apps/web/vibes/soul/primitives/inline-email-form/index.tsx

```diff
@@ -35,15 +40,15 @@ export function InlineEmailForm({
       return parseWithZod(formData, { schema });
     },
     shouldValidate: 'onSubmit',
-    shouldRevalidate: 'onInput',
+    shouldRevalidate: 'onBlur',
   });
 
   useEffect(() => {
-    if (lastResult?.error) {
-      console.log(lastResult.error);
+    if (lastResult?.status === 'success') {
+      toast.success(successMessage, { dismissLabel });
       return;
     }
-  }, [lastResult]);
+  }, [dismissLabel, lastResult, successMessage]);
```

Comment:
I feel like when submission succeeds we should have the success state be inline on the form, like the error instead of using a toast. cc: @andrewreifman. I like what we do in https://vibes.site.

Reply:
I agree but if so lets tackle in a different PR.